coordinator/ethereum/message-anchoring-i2/src/main/kotlin/linea/anchoring/clients/L1MessageSentEventsFetcher.kt

```diff
@@ -0,0 +1,103 @@
+package linea.anchoring.clients
+
+import linea.EthLogsSearcher
+import linea.anchoring.events.L1RollingHashUpdatedEvent
+import linea.anchoring.events.MessageSentEvent
+import linea.domain.BlockParameter
+import linea.domain.BlockParameter.Companion.toBlockParameter
+import linea.domain.EthLogEvent
+import linea.kotlin.toHexStringUInt256
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import kotlin.time.Duration
+
+/**
+ * This class is responsible for fetching L1 MessageSent events from the Ethereum blockchain.
+ * It leverages the invariant of increasing message numbers to efficiently search for events on L1 by never going back
+ *
+ * This is relevant when MessageSent events are very sparse on the chain, it requires 10_000K l1 blocks
+ * to collect 100 messages.
+ */
+internal class L1MessageSentEventsFetcher(
+  private val l1SmartContractAddress: String,
+  private val l1EventsSearcher: EthLogsSearcher,
+  private val highestBlockNumber: BlockParameter
```

Comment:
`BlockNumber` part makes it confusing, because it's going to be a block tag more often. I suggest to rename it either to `blockParameter` or even `blockTag`, which is less precise, but IMO more pragmatic